sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/ByteKeyRangeTracker.java

```diff
@@ -54,8 +60,25 @@ public synchronized ByteKeyRange currentRestriction() {
 
   @Override
   public synchronized ByteKeyRange checkpoint() {
-    checkState(lastClaimedKey != null, "Can't checkpoint before any key was successfully claimed");
-    ByteKey nextKey = next(lastClaimedKey);
+    // If we haven't done any work, we should return the original range we were processing
+    // as the checkpoint.
+    if (lastAttemptedKey == null) {
+      ByteKeyRange rval = ByteKeyRange.of(range.getStartKey(), range.getEndKey());
+      // We update our current range to an interval that contains no elements.
+      range = NO_KEYS;
+      return rval;
```

Comment:
Newbie question: would it be incorrect to return the original `range` instead of creating a new one?

Reply:
Yes, for some reason I goofed and forgot how to do a swap using a temp variable. I'll submit that as a follow up change since it is a minor optimization over the current implementation and the tests passed.